### PR TITLE
[Feature] Support Zeta engine on Kubernetes

### PR DESCRIPTION
**Pull request by xiamidavid00**
Add Doc for Zeta engine running on k8s and add test case for zeta on k8s.

### CODE REVIEW

File: docs/en/start-v2/kubernetes/kubernetes.mdx [hunk 1]
```diff
@@ -15,8 +15,10 @@ We assume that you have a local installations of the following:
 
 - [docker](https://docs.docker.com/)
 - [kubernetes](https://kubernetes.io/)
+If you want to use the Flink engine, you need helm
 - [helm](https://helm.sh/docs/intro/quickstart/)
-
+If you want to use the cluster mode of the Zeta engine, you need HDFS to save Checkpoint
```
Comment (Hisoka-X) [inline docs/en/start-v2/kubernetes/kubernetes.mdx]: I think `HDFS` is a option, we also support s3. And user can do not use save checkpoint feature, like only run batch job.

Comment (xiamidavid00) [reply docs/en/start-v2/kubernetes/kubernetes.mdx]: Okay, I'll adjust it

Comment (xiamidavid00) [reply docs/en/start-v2/kubernetes/kubernetes.mdx]: This is indeed not necessary

File: docs/en/start-v2/kubernetes/kubernetes.mdx [hunk 2]
```diff
@@ -36,6 +36,8 @@ To run the image with SeaTunnel, first create a `Dockerfile`:
   defaultValue="flink"
   values={[
     {label: 'Flink', value: 'flink'},
+    {label: 'zeta (local-mode)', value: 'zeta (local-mode)'},
```
Comment (Carl-Zhou-CN) [inline docs/en/start-v2/kubernetes/kubernetes.mdx]: Would it be better to change the label to 'Zeta'

Comment (xiamidavid00) [reply docs/en/start-v2/kubernetes/kubernetes.mdx]: I'll adjust it

File: docs/en/start-v2/kubernetes/kubernetes.mdx (hunk 1 above)
Comment (xiamidavid00) [reply docs/en/start-v2/kubernetes/kubernetes.mdx]: done

File: docs/en/start-v2/kubernetes/kubernetes.mdx (hunk 2 above)
Comment (xiamidavid00) [reply docs/en/start-v2/kubernetes/kubernetes.mdx]: done

**Comment by Hisoka-X** [comment]
Do we have some way to test running on k8s with code? I'm worried that this document will eventually become invalid as versions change, but we don't know. If we have test cases that run the steps in the documentation, when we need to update the test case in the future, we can update the documentation simultaneously.

File: seatunnel-e2e/seatunnel-engine-e2e/pom.xml [hunk 3]
```diff
@@ -31,6 +31,8 @@
     <modules>
         <module>connector-seatunnel-e2e-base</module>
         <module>connector-console-seatunnel-e2e</module>
+        <module>seatunnel-engine-k8s-e2e</module>
+
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-engine-e2e/pom.xml]: Please remove empty line.

File: docs/en/start-v2/kubernetes/kubernetes.mdx [hunk 4]
```diff
@@ -36,6 +36,8 @@ To run the image with SeaTunnel, first create a `Dockerfile`:
   defaultValue="flink"
```
Comment (Hisoka-X) [inline docs/en/start-v2/kubernetes/kubernetes.mdx]: Let's change defaultValue to `Zeta (local-mode)`!

File: seatunnel-e2e/seatunnel-engine-e2e/seatunnel-engine-k8s-e2e/src/test/java/org/apache/seatunnel/engine/e2e/k8s/KubernetesIT.java [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+        AppsV1Api appsV1Api = new AppsV1Api(client);
+        CoreV1Api coreV1Api = new CoreV1Api(client);
+        MavenXpp3Reader pomReader = new MavenXpp3Reader();
+        Model model = pomReader.read(new FileReader(pomPath), true);
+        String artifactId = model.getArtifactId();
+        String tag = artifactId + ":lastest";
+        Info info = dockerClient.infoCmd().exec();
+        log.info("Docker's environmental information");
+        log.info(info.toString());
+        if (dockerClient.listImagesCmd().withImageNameFilter(tag).exec().size() > 0) {
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-engine-e2e/seatunnel-engine-k8s-e2e/src/test/java/org/apache/seatunnel/engine/e2e/k8s/KubernetesIT.java]:
```suggestion
        if (!dockerClient.listImagesCmd().withImageNameFilter(tag).exec().isEmpty()) {
```

File: seatunnel-e2e/seatunnel-engine-e2e/seatunnel-engine-k8s-e2e/src/test/java/org/apache/seatunnel/engine/e2e/k8s/KubernetesIT.java [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+        copyFileToCurrentResources(targetPath);
+        DockerClient dockerClient = DockerClientFactory.lazyClient();
+        String pomPath = PROJECT_ROOT_PATH + "/pom.xml";
+        ApiClient client = Config.defaultClient();
+        AppsV1Api appsV1Api = new AppsV1Api(client);
+        CoreV1Api coreV1Api = new CoreV1Api(client);
+        MavenXpp3Reader pomReader = new MavenXpp3Reader();
+        Model model = pomReader.read(new FileReader(pomPath), true);
+        String artifactId = model.getArtifactId();
+        String tag = artifactId + ":lastest";
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-engine-e2e/seatunnel-engine-k8s-e2e/src/test/java/org/apache/seatunnel/engine/e2e/k8s/KubernetesIT.java]:
```suggestion
        String tag = artifactId + ":latest";
```

File: seatunnel-e2e/seatunnel-engine-e2e/seatunnel-engine-k8s-e2e/src/test/resources/seatunnel-statefulset.yaml [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    matchLabels:
+      app: seatunnel
+  template:
+    metadata:
+      labels:
+        app: seatunnel
+    spec:
+      containers:
+        - name: seatunnel
+          image: seatunnel:lastest
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-engine-e2e/seatunnel-engine-k8s-e2e/src/test/resources/seatunnel-statefulset.yaml]:
```suggestion
          image: seatunnel:latest
```

File: seatunnel-e2e/seatunnel-engine-e2e/seatunnel-engine-k8s-e2e/src/test/java/org/apache/seatunnel/engine/e2e/k8s/KubernetesIT.java [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+        Thread.sleep(5000);
+        V1StatefulSet v1StatefulSet =
+                appsV1Api.readNamespacedStatefulSet("seatunnel", "default", null);
+        // assert ready replicas
+        Assertions.assertNotNull(v1StatefulSet.getStatus().getReadyReplicas());
+        Assertions.assertEquals(v1StatefulSet.getStatus().getReadyReplicas(), 2);
+        log.info(v1StatefulSet.toString());
+        // submit job
+        String command =
+                "sh /opt/seatunnel/bin/seatunnel.sh --config /opt/seatunnel/config/v2.batch.config.template";
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-engine-e2e/seatunnel-engine-k8s-e2e/src/test/java/org/apache/seatunnel/engine/e2e/k8s/KubernetesIT.java]: I think we should add a negative case too.

**Review by Hisoka-X — DISMISSED** [review]
Thanks @xiamidavid00 ! This is really cool!

**Review by Carl-Zhou-CN — APPROVED** [review]
+1 ,niubility！

File: docs/en/start-v2/kubernetes/kubernetes.mdx [hunk 9]
```diff
@@ -15,8 +15,8 @@ We assume that you have a local installations of the following:
 
 - [docker](https://docs.docker.com/)
 - [kubernetes](https://kubernetes.io/)
+If you want to use the Flink engine, you need helm
```
Comment (Hisoka-X) [inline docs/en/start-v2/kubernetes/kubernetes.mdx]:
```suggestion
```

File: docs/en/start-v2/kubernetes/kubernetes.mdx [hunk 10]
```diff
@@ -16,7 +16,6 @@ We assume that you have a local installations of the following:
 - [docker](https://docs.docker.com/)
 - [kubernetes](https://kubernetes.io/)
 - [helm](https://helm.sh/docs/intro/quickstart/)
-
 So that the `kubectl` and `helm` commands are available on your local system.
```
Comment (Hisoka-X) [inline docs/en/start-v2/kubernetes/kubernetes.mdx]:
```suggestion

So that the `kubectl` and `helm` commands are available on your local system.
```

**Review by Hisoka-X — APPROVED** [review]

**Review by hailin0 — APPROVED** [review]

**Pull request merged**